INFO details about fragmentation

oranagra opened the pull request:
the details about `mem_fragmentation_ratio`, and `activedefrag_running` where misleading.
added lost of info about the breakdown of different RSS related metrics.

huangzhw commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+    Note that this doesn't only includes fragmentation, but also other process overheads (see the `allocator_*` metrics), and also overheads like code, shared libraries, stack, etc.
+*   `mem_fragmentation_bytes`: Delta between `used_memory_rss` and `used_memory`
+    Note that when the total fragmentation bytes is low (few megabytes), a high ratio (e.g. 1.5 and above) is not an indication of an issue.
+*   `allocator_frag_ratio:`: Ratio between `allocator_allocated` and `allocator_active`. This is the true (external) fragmentation metric (not `mem_fragmentation_ratio`)
```
```suggestion
*   `allocator_frag_ratio:`: Ratio between `allocator_active` and `allocator_allocated`. This is the true (external) fragmentation metric (not `mem_fragmentation_ratio`)
```

huangzhw commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+*   `mem_fragmentation_bytes`: Delta between `used_memory_rss` and `used_memory`
+    Note that when the total fragmentation bytes is low (few megabytes), a high ratio (e.g. 1.5 and above) is not an indication of an issue.
+*   `allocator_frag_ratio:`: Ratio between `allocator_allocated` and `allocator_active`. This is the true (external) fragmentation metric (not `mem_fragmentation_ratio`)
+*   `allocator_frag_bytes` Delta between `allocator_allocated` and `allocator_active`. See note about `mem_fragmentation_bytes`.
```
```suggestion
*   `allocator_frag_bytes` Delta between `allocator_active` and `allocator_allocated`. See note about `mem_fragmentation_bytes`.
```

huangzhw commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+    Note that when the total fragmentation bytes is low (few megabytes), a high ratio (e.g. 1.5 and above) is not an indication of an issue.
+*   `allocator_frag_ratio:`: Ratio between `allocator_allocated` and `allocator_active`. This is the true (external) fragmentation metric (not `mem_fragmentation_ratio`)
+*   `allocator_frag_bytes` Delta between `allocator_allocated` and `allocator_active`. See note about `mem_fragmentation_bytes`.
+*   `allocator_rss_ratio`: Ratio between `allocator_active` and `allocator_resident`. This usually indicates pages that the allocator can and probably will soon release back to the OS.
```
```suggestion
*   `allocator_rss_ratio`: Ratio between `allocator_resident` and `allocator_active`. This usually indicates pages that the allocator can and probably will soon release back to the OS.
```

huangzhw commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+*   `allocator_frag_ratio:`: Ratio between `allocator_allocated` and `allocator_active`. This is the true (external) fragmentation metric (not `mem_fragmentation_ratio`)
+*   `allocator_frag_bytes` Delta between `allocator_allocated` and `allocator_active`. See note about `mem_fragmentation_bytes`.
+*   `allocator_rss_ratio`: Ratio between `allocator_active` and `allocator_resident`. This usually indicates pages that the allocator can and probably will soon release back to the OS.
+*   `allocator_rss_bytes`: Delta between `allocator_active` and `allocator_resident`. 
```
```suggestion
*   `allocator_rss_bytes`: Delta between `allocator_resident` and `allocator_active`. 
```

huangzhw commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+*   `allocator_frag_bytes` Delta between `allocator_allocated` and `allocator_active`. See note about `mem_fragmentation_bytes`.
+*   `allocator_rss_ratio`: Ratio between `allocator_active` and `allocator_resident`. This usually indicates pages that the allocator can and probably will soon release back to the OS.
+*   `allocator_rss_bytes`: Delta between `allocator_active` and `allocator_resident`. 
+*   `rss_overhead_ratio`: Ratio between `allocator_resident` and `used_memory_rss` (the process RSS). This includes RSS overheads that are not allocator or heap related.
```
```suggestion
*   `rss_overhead_ratio`: Ratio between `used_memory_rss` (the process RSS) and `allocator_resident`. This includes RSS overheads that are not allocator or heap related.
```

huangzhw commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+*   `allocator_rss_ratio`: Ratio between `allocator_active` and `allocator_resident`. This usually indicates pages that the allocator can and probably will soon release back to the OS.
+*   `allocator_rss_bytes`: Delta between `allocator_active` and `allocator_resident`. 
+*   `rss_overhead_ratio`: Ratio between `allocator_resident` and `used_memory_rss` (the process RSS). This includes RSS overheads that are not allocator or heap related.
+*   `rss_overhead_bytes`: Delta between `allocator_resident` and `used_memory_rss` (the process RSS).
```
```suggestion
*   `rss_overhead_bytes`: Delta between `used_memory_rss` (the process RSS) and `allocator_resident`.
```

huangzhw commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+*   `allocator_resident`: Total bytes resident (RSS) in the allocator, this includes pages that can be released to the OS (by `MEMORY PURGE`, or just waiting)
 *   `mem_allocator`: Memory allocator, chosen at compile time
-*   `active_defrag_running`: Flag indicating if active defragmentation is active
+*   `active_defrag_running`: When `activedefrag` is enabled, this indicates if active if is active, and the CPU percentage it intends to utilize
```
```suggestion
*   `active_defrag_running`: When `activedefrag` is enabled, this indicates whether active defrag is active, and the CPU percentage it intends to utilize
```

oranagra commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+*   `allocator_resident`: Total bytes resident (RSS) in the allocator, this includes pages that can be released to the OS (by `MEMORY PURGE`, or just waiting)
 *   `mem_allocator`: Memory allocator, chosen at compile time
-*   `active_defrag_running`: Flag indicating if active defragmentation is active
+*   `active_defrag_running`: When `activedefrag` is enabled, this indicates whether active defrag is active, and the CPU percentage it intends to utilize
```
```suggestion
*   `active_defrag_running`: When `activedefrag` is enabled, this indicates whether defragmentation is currently active, and the CPU percentage it intends to utilize
```

yoav-steinberg commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+*   `rss_overhead_bytes`: Delta between `used_memory_rss` (the process RSS) and `allocator_resident`.
+*   `allocator_allocated`: Total bytes allocated form the allocator, including internal-fragmentation. Normally the same as `used_memory`.
+*   `allocator_active`: Total bytes in the allocator active pages, this includes external-fragmentation.
+*   `allocator_resident`: Total bytes resident (RSS) in the allocator, this includes pages that can be released to the OS (by `MEMORY PURGE`, or just waiting)
```
```suggestion
*   `allocator_resident`: Total bytes resident (RSS) in the allocator, this includes pages that can be released to the OS (by `MEMORY PURGE`, or just waiting).
```

yoav-steinberg commented on commands/info.md:
```diff
@@ -122,18 +122,29 @@ Here is the meaning of all fields in the **memory** section:
+*   `allocator_resident`: Total bytes resident (RSS) in the allocator, this includes pages that can be released to the OS (by `MEMORY PURGE`, or just waiting)
 *   `mem_allocator`: Memory allocator, chosen at compile time
-*   `active_defrag_running`: Flag indicating if active defragmentation is active
+*   `active_defrag_running`: When `activedefrag` is enabled, this indicates whether defragmentation is currently active, and the CPU percentage it intends to utilize
```
```suggestion
*   `active_defrag_running`: When `activedefrag` is enabled, this indicates whether defragmentation is currently active, and the CPU percentage it intends to utilize.
```

yoav-steinberg reviewed (approved):
missing periods. In general in lists either all items should end with a period or none should.

oranagra commented:
@yoav-steinberg i think i noticed that most of the entries in that list are without a period.
so maybe the problem is that in several of the new ones i added there is one, and instead of adding the 2 you suggested we want to remove the other ones i added?
